pyani_plus/public_cli.py

```diff
@@ -870,5 +859,82 @@ def plot_run(  # noqa: C901
     return 0
 
 
+@app.command()
+def classify(  # noqa: PLR0913
+    database: REQ_ARG_TYPE_DATABASE,
+    outdir: REQ_ARG_TYPE_OUTDIR,
+    covearge_edges: Annotated[
+        str,
+        typer.Option(
+            help="How to resolve asymmetrical ANI identity results for edges in the graph (min, max or mean).",
+            rich_help_panel="Method parameters",
+        ),
+    ] = "min",  # Default to "min"
+    identity_edges: Annotated[
+        str,
+        typer.Option(
+            help="How to resolve asymmetrical ANI identity results for edges in the graph (min, max or mean).",
+            rich_help_panel="Method parameters",
+        ),
+    ] = "mean",  # Default to "mean"
+    run_id: OPT_ARG_TYPE_RUN_ID = None,
+    label: OPT_ARG_TYPE_LABEL = "stem",
+) -> int:
+    """Classify genomes into clusters based on ANI results."""
+    if not outdir.is_dir():
+        msg = f"ERROR: Output directory {outdir} does not exist"
+        sys.exit(msg)
+
+    if database == ":memory:" or not Path(database).is_file():
+        msg = f"ERROR: Database {database} does not exist"
+        sys.exit(msg)
+
+    session = db_orm.connect_to_db(database)
+    run = db_orm.load_run(session, run_id, check_complete=True)
+    if run_id is None:
+        run_id = run.run_id
+        print(f"INFO: Exporting run-id {run_id}")
+
+    method = run.configuration.method
+
+    identity = run.identities
+    if identity is None:
+        msg = f"ERROR: Could not load run {method} matrix"  # pragma: no cover
+        sys.exit(msg)  # pragma: no cover
+
+    cov = run.cov_query
+    if cov is None:
+        msg = f"ERROR: Could not load run {method} matrix"  # pragma: no cover
+        sys.exit(msg)  # pragma: no cover
+
+    try:
+        identity = run.relabelled_matrix(identity, label)
+        cov = run.relabelled_matrix(cov, label)
+    except ValueError as err:
+        msg = f"ERROR: {err}"
+        sys.exit(msg)
```

Comment:
This exception isn't currently covered in the tests. You can just add a call to your new public API in the ``test_export_duplicate_stem`` function.